pkgs/development/libraries/modern-cpp-kafka/default.nix

```diff
@@ -0,0 +1,48 @@
+{ rdkafka, stdenv, lib, fetchFromGitHub, cmake, zlib, breakpointHook, boost, gtest, rapidjson, pkg-config }:
+
+stdenv.mkDerivation rec {
+  pname = "modern-cpp-kafka";
+  version = "v2023.03.07";
+
+  src = fetchFromGitHub {
+    repo = "modern-cpp-kafka";
+    # owner = "morganstanley";
+    # rev = version;
+    # hash = "sha256-7hkwM1YbveQpDRqwMZ3MXM88LTwlAT7uB8NL0t409To=";
+    #
+    # Patch build to avoid overwriting include path
+    # PR: https://github.com/morganstanley/modern-cpp-kafka/pull/221
+    owner = "ditsuke";
+    rev = "build/avoid-overwriting-include-path";
+    hash = "sha256-fRKIDaDkjBHKKrBUj/TfjQpvF1jMQacj5AXB28cLWxE=";
+  };
+
+  cmakeFlags = [
+    "-DLIBRDKAFKA_INCLUDE_DIR=${rdkafka.out}/include"
+    "-DGTEST_LIBRARY_DIR=${gtest.out}/lib"
+    "-DGTEST_INCLUDE_DIR=${gtest.dev}/include"
+    "-DRAPIDJSON_INCLUDE_DIRS=${rapidjson.out}/include"
+    "-DCMAKE_CXX_FLAGS=-Wno-uninitialized"
+  ];
+
+  buildInputs = [ boost ];
+  propagatedBuildInputs = [ rdkafka ];
+  testInputs = [ gtest rapidjson ];
+  nativeBuildInputs = [ cmake ];
+
+  buildPhase = ''
+    cmake .
```

Comment:
Calls to `cmake`, which is a configuration tool, do not belong into the buildPhase.
Do you intent to just provide the headers with this library, or do you also plan to ship compiled library files also? For only header files, you may not even need a configuration phase. But that should be handled by having `cmake` in `nativeBuildInputs` already.

Reply:
Just the headers